src/libraries/System.Linq/src/System/Linq/First.cs

```diff
@@ -31,12 +31,21 @@ public static TSource First<TSource>(this IEnumerable<TSource> source, Func<TSou
         }
 
         public static TSource? FirstOrDefault<TSource>(this IEnumerable<TSource> source) =>
-            source.TryGetFirst(out bool _);
+            source.TryGetFirst(out _);
+
+        public static TSource FirstOrDefault<TSource>(this IEnumerable<TSource> source, TSource defaultValue) =>
+            source.TryGetFirst(defaultValue, out _)!;
 
         public static TSource? FirstOrDefault<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate) =>
-            source.TryGetFirst(predicate, out bool _);
+            source.TryGetFirst(predicate, out _);
+
+        public static TSource FirstOrDefault<TSource>(this IEnumerable<TSource> source, Func<TSource, bool> predicate, TSource defaultValue) =>
+            source.TryGetFirst(predicate, defaultValue, out _)!;
 
-        private static TSource? TryGetFirst<TSource>(this IEnumerable<TSource> source, out bool found)
+        private static TSource? TryGetFirst<TSource>(this IEnumerable<TSource> source, out bool found) =>
```

Comment:
I'd probably just get rid of that method and call the core implementation directly.